qutip/core/cy/coefficient.pyx

```diff
@@ -164,6 +179,12 @@ cdef class InterpolateCoefficient(Coefficient):
     def __reduce__(self):
         return InterpolateCoefficient, (self.spline,)
 
+    cpdef Coefficient copy(self):
+        return InterpolateCoefficient(self.spline)
+
+    def replace(self, *, arguments=None, tlist=None):
+        return self
```

Comment:
This seems somewhat misleading; shouldn't we make a new spline object if the `tlist` is updated?

Reply:
Also, what's the difference between `InterpolateCoefficient` and `InterCoefficient`?  Could we have some class docstrings on all the `Coefficient` objects to make it clear too?

Reply:
Cannot be done easily, the original coefficient array is not stored in the Cubic_Spline object.
`Inter` and `Interpolate` are 2 implementations of cubic spline.
We should deprecate one of them with v5, now that we can.
`Cubic_Spline` is more memory efficient, `InterCoefficient` is faster.
`Cubic_Spline` support setting second derivative at boundaries, `InterCoefficient` support non-uniform `tlist`.